cirq/ops/eigen_gate.py

```diff
@@ -59,7 +60,7 @@ class EigenGate(raw_types.Gate):
     """
 
     def __init__(self, *,  # Forces keyword args.
-                 exponent: Union[value.Symbol, float] = 1.0,
+                 exponent: Union[sympy.Symbol, float] = 1.0,
```

Comment:
I suspect a lot of the `sympy.Symbol` types should change to `sympy.Basic`, because they can be formulae. For now it's fine to leave at as symbol and ease in.